src/sentry/snuba/discover.py

```diff
@@ -1060,6 +1060,8 @@ def find_histogram_min_max(fields, min_value, max_value, user_query, params, dat
         min_values = [row[get_function_alias(column)] for column in min_columns]
         min_values = list(filter(lambda v: v is not None, min_values))
         min_value = min(min_values) if min_values else None
+        if max_value is not None and min_value is not None:
+            min_value = min([max_value, min_value])
```

Comment:
@Zylphrex I've added a counterpart for ensuring the queried min value is consistent with respect to max value. This would've cropped up as another bug in the future. Squashing it here so it doesn't ever occur. I've added the corresponding tests to cover this case.

Reply:
I agree this should be addressed though I'm not sure this is the right approach here.

The histogram endpoint should always respect the `min` value (rounded to the nearest nice number) while the `max` value should be taken as a suggestion (i.e. outlier filtering may kick in).

By doing what you have here, you're lowering the min value, thus including even more values than expected in the results.

To address this, I think what should be done is to add a check after we finish computing both the `min_value` and `max_value`. And we can do something like

```python
if min_value is not None and max_value is not None:
    max_value = max(min_value, max_value)
```

to respect the `min_value` as much as possible while raising the `max_value` to satisfy the invariant.

Reply:
@Zylphrex To clarify my understanding, if only the `max_value` is given by the user, are we allowed to mutate this value with respect to the queried `min` value?

Reply:
Okay, as discussed offline, this looks good and does give reasonable results to the user.